packages/form-core/src/registration/FormRegistrarMixin.js

```diff
@@ -154,10 +155,16 @@ const FormRegistrarMixinImplementation = superclass =>
 
       // 2. Add children as object key
       if (this._isFormOrFieldset) {
-        const { name } = child;
+        let { name } = child;
         if (!name) {
-          console.info('Error Node:', child); // eslint-disable-line no-console
-          throw new TypeError('You need to define a name');
+          // Give child a chance to render if property bindings are done later
+          // E.g. for @lit-labs/react
+          await child.updateComplete;
```

Comment:
Sorry for the late feedback (totally occupied with smth else back then and forgot about this one)....

So the goal is to prevent React from throwing name errors? 
Maybe we can always wait for the updateComplete (via .then) and afterwards throw the error.
Allows us to keep the method sync (since I'm afraid it can lead to nasty timing errors in our extension layers otherwise)